src/asqi/workflow.py

```diff
@@ -423,15 +431,34 @@ def run_test_suite_workflow(
     with console.status("[bold blue]Checking image availability...", spinner="dots"):
         image_availability = dbos_check_images_availability(unique_images)
 
+    # Try to pull missing images from Docker Hub/registries
     missing_images = [
         img for img, available in image_availability.items() if not available
     ]
     if missing_images:
         console.print(
             f"[yellow]Warning:[/yellow] {len(missing_images)} images not available locally"
         )
+        with console.status(
+            "[bold blue]Pulling missing images from registry...", spinner="dots"
+        ):
+            pull_results = dbos_pull_images(missing_images)
+            # Merge pull results into availability map
+            for img, pulled in pull_results.items():
```

Comment:
`pull_results` over here would be `None`. I think you might need to update `pull_images` to return a dict instead.